### PR TITLE
Set up CI

### PR DESCRIPTION
**Pull request by dprokoptsev**
Set up a github action to build and run tests.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
+name: Build and run tests
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
Comment (oremanj) [inline .github/workflows/ci.yml:7]: I don't think you want to condition PRs like this; they generally won't be on the master branch. You can just write `pull_request:` with nothing after it.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          - variant: gcc-aarch64
+            deps: g++-12-aarch64-linux-gnu qemu-user-static
+            cc: aarch64-linux-gnu-gcc-12
+            cxx: aarch64-linux-gnu-g++-12
+            cxxflags: -static
+            test_runner: qemu-aarch64-static
+            
+          - variant: msvc-wine
+
+            # Note: windows workers are in short supply on github,
```
Comment (oremanj) [inline .github/workflows/ci.yml:59]: They're not in that short of supply; I include Windows in the CI matrix for several of my python projects and haven't noticed more than a couple minutes' delay in practice.

**Review by oremanj — APPROVED** [review]
Some tweaks suggested, but looks good!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dprokoptsev) [reply .github/workflows/ci.yml:59]: Mine have not been starting for half an hour, so I ran out of patience.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dprokoptsev) [reply .github/workflows/ci.yml]: The checks did trigger on this PR didn't they?
Fixed nevertheless.

Comment (oremanj) [reply .github/workflows/ci.yml]: Ah huh, I guess it's referring to the branch the PR wants to merge into then. So feel free to go either way, I was confused.

**Pull request merged**